ibc-testkit/src/testapp/ibc/clients/mock/client_state.rs

```diff
@@ -427,6 +432,45 @@ where
 
         Ok(latest_height)
     }
+
+    fn update_on_recovery(
+        &self,
+        ctx: &mut E,
+        subject_client_id: &ClientId,
+        substitute_client_state: Any,
+    ) -> Result<(), ClientError> {
+        let substitute_client_state =
+            TryInto::<ClientStateType>::try_into(substitute_client_state)?;
```

Comment:
This is trying to substitute Mock client state with Tendemint client state. Is this intended?

Reply:
Now that I think about it, I'm not entirely clear on how client recovery should be implemented by the mock module. Converting the substitute client into a Tendermint client state is necessary if we want to mock up storing the update metadata in the store. But if doing that is not necessary for the mock implementation, then I guess we can convert the substitute client state to a mock client state instead. 

Curious what @Farhad-Shabani thinks is appropriate here.

Reply:
The `MockClientState` has the header of the latest height. So the implementation should be something like this:

```rs
       let substitute_client_state = MockClientState::try_from(substitute_client_state)?;

        let MockClientState {
            header,
            trusting_period,
            ..
        } = substitute_client_state;

        let latest_height = header.height();

        let new_mock_client_state = MockClientState {
            frozen: false,
            header,
            trusting_period,
        };
```

Reply:
[dc0a3aa](https://github.com/cosmos/ibc-rs/pull/1127/commits/dc0a3aa6386d013f2bd89686a15ad973b4e80836)

Reply:
Small nit: You can write `new_mock_client_state` as 

```rs
        let new_mock_client_state = MockClientState {
            frozen: false,
            ..substitute_client_state,
        };
```